packages/vite/src/node/optimizer/scan.ts

```diff
@@ -70,9 +84,23 @@ export function scanImports(config: ResolvedConfig): {
   let entries: string[]
 
   const scanContext = { cancelled: false }
+  const scannableExtensions = config.optimizeDeps.extensions
+    ? [
+        ...new Set(
+          defaultScannableExtensions.concat(
+            config.optimizeDeps.extensions.map((ext) => ext.slice(1)),
+          ),
+        ),
+      ]
+    : defaultScannableExtensions
+  const scannableExtensionsRE = new RegExp(
+    `\\.(${scannableExtensions.join('|')})$`,
+  )
+  const isScannable = (id: string) => scannableExtensionsRE.test(id)
```

Comment:
I was thinking the previous commit you made was already great 😅 (Passing `config.optimizeDeps.extensions` to `isScannable`)

I think this new change duplicates `scannableExtensions` which will be another thing to maintain.

It should be fine to do `extensions.includes(path.extname(id))` though, since `JS_TYPES_RE` and `htmlTypesRE` covers many of the best case by default.

Reply:
Alright I can revert to that, one moment!